### DISCUSSION TITLE
Can i use OrchardCore.Tenants module in my project

### DISCUSSION THREAD

**Discussion by kfathy-ys** [OrchardCMS/OrchardCore#11449] [Q&A] [answered]
Can I use OrchardCore.Tenants in my project and make my CRUD ui using angualr that use manage tenent and control how to store in my data base and make my entityframwork linq queries aware by current tenent without developer care about?

**Comment by Skrypt** [comment]
I would say it is not encouraged to have your own DAL. We have plenty of API's to use Angular with. Also, you can create your own REST API by using examples we have in the source code. But yes, everything is possible.

**Reply by kfathy-ys** [reply]
Thanks for replay , i cannot find any example to handle tenant crud and save it in databse , the examples only explain how to set tenents in appsetting.json
Can you send me any example.
Also please explain more about the sentence 'We have plenty of API's to use Angular with'

**Comment by ns8482e** [comment, marked as answer]
Short answer - Yes you can create your on tenant management UI

`OrchardCore.Tenants` is just a tenant management UI provided out of the box

**Reply by kfathy-ys** [reply]
Thanks for replay , i cannot find any example to handle tenant crud and save it in databse , the examples only explain how to set tenents in appsetting.json
Can you send me any example.

**Reply by ns8482e** [reply]
@kfathy-ys why example? Just Look at source in AdminController in tenant module. You can also look at code for  OrchardCore.Try repo. 

`Tenant` is not just data record, it's also an `Infrastructure` and  default implementation of `OrchardCore.Tenants` directly operates multi-tenant infrastructure configuration items like  `tenants.json`, site folder etc.  -  that you still need regardless if you store a copy in db.
